### PR TITLE
fix(terraform): externalhosterror for 503

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Throws ExternalHostError for terraform 503

## Context:

Autocloses in the app

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/terraform-module/index.ts [hunk 1]
```diff
@@ -89,6 +89,10 @@ export class TerraformModuleDatasource extends TerraformDatasource {
     if (failureCodes.includes(err.code)) {
       throw new ExternalHostError(err);
     }
+    // istanbul ignore if
+    if (err.response?.statusCode === 503) {
```
Comment (viceice) [inline lib/datasource/terraform-module/index.ts:93]: Why not add to failure codes above?

File: lib/datasource/terraform-provider/index.ts [hunk 2]
```diff
@@ -63,6 +64,18 @@ export class TerraformProviderDatasource extends TerraformDatasource {
...
   }
 
+  override handleSpecificErrors(err: HttpError): void {
+    const failureCodes = ['EAI_AGAIN'];
+    // istanbul ignore if
+    if (failureCodes.includes(err.code)) {
+      throw new ExternalHostError(err);
+    }
+    // istanbul ignore if
+    if (err.response?.statusCode === 503) {
```
Comment (viceice) [inline lib/datasource/terraform-provider/index.ts:74]: Add this to failure codes instead?

File: lib/datasource/terraform-module/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/datasource/terraform-module/index.ts:93]: I think status code is different to err code

Comment (viceice) [reply lib/datasource/terraform-module/index.ts:93]: ok, you're right

File: lib/datasource/terraform-provider/index.ts [hunk 3]
```diff
@@ -63,6 +64,18 @@ export class TerraformProviderDatasource extends TerraformDatasource {
     return dep;
   }
 
+  override handleSpecificErrors(err: HttpError): void {
```
Comment (viceice) [inline lib/datasource/terraform-provider/index.ts:67]: Needs a test. 😏

Comment (JamieMagee) [reply lib/datasource/terraform-provider/index.ts:67]: `TerraformProviderDatasource` and `TerraformModuleDatasource` share the same `handleSpecificErrors`. You could move this to `TerraformDatasource`.

Comment (viceice) [reply lib/datasource/terraform-provider/index.ts:67]: Good idea 👍

File: lib/datasource/terraform-module/base.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
+import { ExternalHostError } from '../../types/errors/external-host-error';
 import { cache } from '../../util/cache/package/decorator';
+import { HttpError } from '../../util/http/types';
```
Comment (viceice) [inline lib/datasource/terraform-module/base.ts:3]:
```suggestion
import type { HttpError } from '../../util/http/types';
```

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**